domain/src/main/java/org/oppia/android/domain/topic/StoryProgressController.kt

```diff
@@ -113,14 +113,23 @@ class StoryProgressController @Inject constructor(
         storyProgressBuilder.putChapterProgress(explorationId, chapterProgress)
         val storyProgress = storyProgressBuilder.build()
 
-        val topicProgressBuilder =
-          TopicProgress.newBuilder()
-            .setTopicId(topicId).setLastPlayedTimestamp(completionTimestamp)
+        val topicProgressBuilder = TopicProgress.newBuilder().setTopicId(topicId)
+
         if (topicProgressDatabase.topicProgressMap[topicId] != null) {
           topicProgressBuilder
             .putAllStoryProgress(topicProgressDatabase.topicProgressMap[topicId]!!.storyProgressMap)
         }
         topicProgressBuilder.putStoryProgress(storyId, storyProgress)
+        // Compute lastPlayedTimeStamp from chapterProgress list.
+        val topicProgressStories = topicProgressBuilder.storyProgressMap.values
+        val topicProgressChapters = topicProgressStories.flatMap { it.chapterProgressMap.values }
+        val topicProgressLastPlayedTimes =
+          topicProgressChapters.map(ChapterProgress::getLastPlayedTimestamp)
+        val computedLastPlayedTimestamp =
+          topicProgressLastPlayedTimes.max()
```

Comment:
This isn't quite what I meant. We should be computing the last played time stamp where we need to read it (meaning we should remove it from the TopicProgress proto entirely).

Reply:
Ah, yeah done. Removed it from proto.